admin_guide/taints_tolerations.adoc

```diff
@@ -0,0 +1,350 @@
+
+To configure a node so that users can use only that node:
+
+. Add a taint to those nodes: 
++
+---- 
+$ oadm taint nodes nodename dedicated=groupName:NoSchedule`
+----
+
+. Add a corresponding toleration to the pods. This would be done most easily by writing a custom
+xref:../install_config/master_node_configuration.adoc#master-config-admission-control-config[admission controller].
++
+The admission
+controller should additionally add a node affinity to require that the pods can only schedule
+onto nodes labeled with `dedicated=groupName`.
+
+. Add a label similar to the taint to the same set of nodes (such as `dedicated=groupName`). 
+
+
+[[admin-guide-taints-special]]
+### Nodes with Special Hardware 
+
+In a cluster where a small subset of nodes have specialized
+hardware (for example GPUs), you can use taints and tolerations to keep pods that do not need the specialized
+hardware off of those nodes, leaving the nodes for pods that do need the
+specialized hardware. And, you can require pods that need specialized hardware to use specific nodes.   
+
+To make sure pods are blocked from the specialized hardware:
+
+. Taint the nodes that have the specialized
+hardware using one of the following commands:
++
+----
+$ oadm taint nodes nodename special=true:NoSchedule
+$ oadm taint nodes nodename special=true:PreferNoSchedule
+----
+
+. Adding a corresponding
+toleration to pods that use the special hardware using an
+xref:../../architecture/additional_concepts/admission_controllers.html[admission controller].
+
+For example, the admission controller could use
+some characteristic(s) of the pod to determine that the pod should be allowed to use
+the special nodes by adding a toleration.
+
+To make sure pods can only use the specialized hardware, you will need some
+additional mechanism. For example, you could represent the special resource 
+using link:https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#opaque-integer-resources-alpha-feature[opaque integer resources] 
+and request it as a resource in the PodSpec, or you could label the nodes that have
+the special hardware and use node affinity on the pods that need the hardware.
+
+
+## Configuring Per-Pod Eviction (Taint Based Evictions alpha feature)
+
+In addition to the `NoExecute` effect, Kubernetes 1.6 has alpha
+support for representing node problems (currently only *node unreachable* and
+*node not ready*) as taints.  This allows per-pod specification of how long to remain bound to a node 
+that becomes unreachable or not ready, rather than using the default of 5 minutes.
+
+When the Taint Based Evictions alpha feature
+is enabled, the taints are automatically added by the NodeController and the normal logic for evicting pods from nodes
+based on the Ready NodeCondition is disabled.
+
+To enable Taint Based Evictions:
+
+----
+$ kubelet --feature-gates TaintBasedEvictions=true`), the taints are automatically
```

Comment:
Enable TaintBasedEviction should in kube-controller-manager not kubelet, In openshift we can enalbe `master-config.yaml`
```
kubernetesMasterConfig:
   controllerArguments:
        feature-gates:
        - "TaintBasedEvictions=true"
```